addons/stock/models/stock_move.py

```diff
@@ -255,10 +255,19 @@ def _quantity_done_compute(self):
         field will be used in `_action_done` in order to know if the move will need a backorder or
         an extra move.
         """
+        MoveLine = self.env['stock.move.line']
+        ProductUom = self.env['product.uom']
         for move in self:
-            quantity_done = 0
-            for move_line in move._get_move_lines():
-                quantity_done += move_line.product_uom_id._compute_quantity(move_line.qty_done, move.product_uom, round=False)
+            uom = move.product_uom
+            move_lines = MoveLine.read_group(
+                [('id', 'in', move._get_move_lines().ids)],
+                ['product_uom_id', 'qty_done'],
+                ['product_uom_id']
+            )
+            quantity_done = sum(
+                ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
+                for vals in move_lines
+            )
```

Comment:
You can improve again the `read_group`, `read_group` also by `move_id` and with `lazy=False`.

```python
move_lines = self.env['stock.move.line']
for move in self.with_context(prefetch_fields=False):
    move_lines |= move._get_move_lines()

data = MoveLine.read_group([('id', 'in', move_lines)],
['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'], lazy=False])

rec = defaultdict(list)

for d in data:
    rec[d['move_id][0]] += (d['product_uom_id'][0], d['qty_done'])

for move in self:
    uom = move.product_uom
    move.quantity_done = sum(ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False) for line_uom_id, qty in rec.get(move.id, [])
```

Reply:
I open manufacturing order which is done
Finished product tracking by serial number
Qty 1000
42 component by BOM

It is the best result of 5 tries and I function was called 3 times per one open action


Result from odoo profiler with my commit
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
2019-07-04 10:11:26,027 5224 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.05          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.1               MoveLine = self.env['stock.move.line']
1         0         0.07              ProductUom = self.env['product.uom']
44        0         0.52              for move in self:
42        2         113.74                uom = move.product_uom
42        65        2391.54               move_lines = MoveLine.read_group([('id', 'in', move._get_move_lines().ids)], ['product_uom_id', 'qty_done'], ['product_uom_id'])
42        0         0.21                  quantity_done = sum(
42        0         0.15                      ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
42        1         6.47                      for vals in move_lines
                                          )
42        0         22.57                 move.quantity_done = quantity_done

Total:
42        68        2535.42   


2019-07-04 10:11:26,934 5224 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.01              ProductUom = self.env['product.uom']
4         0         0.03              for move in self:
2         2         12.31                 uom = move.product_uom
2         6         885.84                move_lines = MoveLine.read_group([('id', 'in', move._get_move_lines().ids)], ['product_uom_id', 'qty_done'], ['product_uom_id'])
2         0         0.01                  quantity_done = sum(
2         0         0.01                      ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
2         0         0.27                      for vals in move_lines
                                          )
2         0         1.17                  move.quantity_done = quantity_done

Total:
2         8         899.69    

2019-07-04 10:11:29,494 5224 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.01              ProductUom = self.env['product.uom']
42        0         0.48              for move in self:
40        0         1.05                  uom = move.product_uom
40        64        2359.48               move_lines = MoveLine.read_group([('id', 'in', move._get_move_lines().ids)], ['product_uom_id', 'qty_done'], ['product_uom_id'])
40        0         0.21                  quantity_done = sum(
40        0         0.15                      ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
40        0         4.45                      for vals in move_lines
                                          )
40        0         22.04                 move.quantity_done = quantity_done

Total:
40        64        2387.9 

Result from odoo profiler with your improve
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
2019-07-04 10:08:54,830 5037 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
                                      # for move in self:
                                      #     uom = move.product_uom
                                      #     move_lines = MoveLine.read_group([('id', 'in', move._get_move_lines().ids)], ['product_uom_id', 'qty_done'], ['product_uom_id'])
                                      #     quantity_done = sum(
                                      #         ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
                                      #         for vals in move_lines
                                      #     )
                                      #     move.quantity_done = quantity_done
                                  
1         0         0.01              move_lines = self.env['stock.move.line']
44        0         0.73              for move in self.with_context(prefetch_fields=False):
42        23        2787.71               move_lines |= move._get_move_lines()
                                  
1         0         0.01              data = MoveLine.read_group(
1         0         5.53                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'],
1         0         0.0                   ['move_id', 'product_uom_id'],
1         8         238.08                lazy=False
                                      )
1         0         0.01              rec = defaultdict(list)
42        0         0.17              for d in data:
41        0         0.18                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                                  
44        0         0.55              for move in self:
42        2         102.16                uom = move.product_uom
42        0         0.18                  move.quantity_done = sum(
42        0         0.17                      ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False) for line_uom_id, qty in
42        1         30.17                     rec.get(move.id, [])
                                          )

Total:
42        34        3165.73   


2019-07-04 10:08:55,757 5037 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
                                      # for move in self:
                                      #     uom = move.product_uom
                                      #     move_lines = MoveLine.read_group([('id', 'in', move._get_move_lines().ids)], ['product_uom_id', 'qty_done'], ['product_uom_id'])
                                      #     quantity_done = sum(
                                      #         ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
                                      #         for vals in move_lines
                                      #     )
                                      #     move.quantity_done = quantity_done
                                  
1         0         0.01              move_lines = self.env['stock.move.line']
4         0         0.1               for move in self.with_context(prefetch_fields=False):
2         6         876.31                move_lines |= move._get_move_lines()
                                  
1         0         0.01              data = MoveLine.read_group(
1         0         0.29                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'],
1         0         0.0                   ['move_id', 'product_uom_id'],
1         6         26.46                 lazy=False
                                      )
1         0         0.01              rec = defaultdict(list)
3         0         0.01              for d in data:
2         0         0.01                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                                  
4         0         0.04              for move in self:
2         2         12.97                 uom = move.product_uom
2         0         0.01                  move.quantity_done = sum(
2         0         0.01                      ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False) for line_uom_id, qty in
2         0         1.31                      rec.get(move.id, [])
                                          )

Total:
2         14        917.62    

2019-07-04 10:08:58,677 5037 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
                                      # for move in self:
                                      #     uom = move.product_uom
                                      #     move_lines = MoveLine.read_group([('id', 'in', move._get_move_lines().ids)], ['product_uom_id', 'qty_done'], ['product_uom_id'])
                                      #     quantity_done = sum(
                                      #         ProductUom.browse(vals['product_uom_id'][0])._compute_quantity(vals['qty_done'], uom, round=False)
                                      #         for vals in move_lines
                                      #     )
                                      #     move.quantity_done = quantity_done
                                  
1         0         0.01              move_lines = self.env['stock.move.line']
42        0         0.65              for move in self.with_context(prefetch_fields=False):
40        24        2571.46               move_lines |= move._get_move_lines()
                                  
1         0         0.01              data = MoveLine.read_group(
1         0         4.91                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'],
1         0         0.0                   ['move_id', 'product_uom_id'],
1         1         75.41                 lazy=False
                                      )
1         0         0.01              rec = defaultdict(list)
40        0         0.15              for d in data:
39        0         0.17                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                                  
42        0         0.49              for move in self:
40        0         1.08                  uom = move.product_uom
40        0         0.15                  move.quantity_done = sum(
40        0         0.15                      ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False) for line_uom_id, qty in
40        0         24.42                     rec.get(move.id, [])
                                          )

Total:
40        25        2679.13

Reply:
Can you profile _get_move_lines(), it is the key of speed. Try without `prefetch_fields`.

Reply:
without prefecth:
```suggestion
2019-07-04 17:09:55,467 18488 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
                                      
1         0         0.01              move_lines = self.env['stock.move.line']
44        0         0.73              for move in self.with_context(prefetch_fields=False):
42        23        2813.21               move_lines |= move._get_move_lines()
                              
1         0         0.01              data = MoveLine.read_group(
1         0         5.66                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'],
1         8         238.97                lazy=False
                                      )
                              
1         0         0.01              rec = defaultdict(list)
                              
42        0         0.17              for d in data:
41        0         0.19                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                              
44        0         0.42              for move in self:
42        2         102.1                 uom = move.product_uom
1         0         0.0               move.quantity_done = sum(
1         0         0.0                   ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False)
1         1         3.18                  for line_uom_id, qty in rec.get(move.id, [])
                                      )

Total:
1         34        3164.72   


2019-07-04 17:09:56,404 18488 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
                                      
1         0         0.01              move_lines = self.env['stock.move.line']
4         0         0.11              for move in self.with_context(prefetch_fields=False):
2         6         887.35                move_lines |= move._get_move_lines()
                              
1         0         0.0               data = MoveLine.read_group(
1         0         0.29                  [('id', 'in', move_lines.ids)],
1         0         0.0                   ['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'],
1         6         26.9                  lazy=False
                                      )
                              
1         0         0.01              rec = defaultdict(list)
                              
3         0         0.01              for d in data:
2         0         0.01                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                              
4         0         0.04              for move in self:
2         2         12.74                 uom = move.product_uom
1         0         0.0               move.quantity_done = sum(
1         0         0.0                   ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False)
1         0         0.68                  for line_uom_id, qty in rec.get(move.id, [])
                                      )

Total:
1         14        928.24    


2019-07-04 17:09:59,290 18488 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
                                      
1         0         0.01              move_lines = self.env['stock.move.line']
42        0         0.72              for move in self.with_context(prefetch_fields=False):
40        24        2538.21               move_lines |= move._get_move_lines()
                              
1         0         0.0               data = MoveLine.read_group(
1         0         5.28                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'],
1         1         78.94                 lazy=False
                                      )
                              
1         0         0.01              rec = defaultdict(list)
                              
40        0         0.16              for d in data:
39        0         0.21                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                              
42        0         0.38              for move in self:
40        0         0.88                  uom = move.product_uom
1         0         0.0               move.quantity_done = sum(
1         0         0.0                   ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False)
1         0         0.69                  for line_uom_id, qty in rec.get(move.id, [])
                                      )

Total:
1         25        2625.55   
```

Reply:
profile _quantity_done_compute and  _get_move_lines without prefetch 
```suggestion
2019-07-04 17:26:12,646 19288 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
1         0         0.01              move_lines = self.env['stock.move.line']
44        0         0.7               for move in self.with_context(prefetch_fields=False):
42        23        2750.59               move_lines |= move._get_move_lines()
                              
1         0         0.01              data = MoveLine.read_group(
1         0         5.49                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'],
1         8         235.61                lazy=False
                                      )
                              
1         0         0.01              rec = defaultdict(list)
                              
42        0         0.17              for d in data:
41        0         0.2                   rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                              
44        0         0.41              for move in self:
42        2         104.33                uom = move.product_uom
1         0         0.0               move.quantity_done = sum(
1         0         0.01                  ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False)
1         1         2.97                  for line_uom_id, qty in rec.get(move.id, [])
                                      )

Total:
1         34        3100.56   


2019-07-04 17:26:13,529 19288 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 243

1         0         0.01          @profile
                                  def _get_move_lines(self):
                                      """ This will return the move lines to consider when applying _quantity_done_compute on a stock.move. 
                                      In some context, such as MRP, it is necessary to compute quantity_done on filtered sock.move.line."""
1         0         0.01              self.ensure_one()
1         4         869.81            return self.move_line_ids or self.move_line_nosuggest_ids

Total:
1         4         869.82    


2019-07-04 17:26:13,530 19288 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 243

1         0         0.01          @profile
                                  def _get_move_lines(self):
                                      """ This will return the move lines to consider when applying _quantity_done_compute on a stock.move. 
                                      In some context, such as MRP, it is necessary to compute quantity_done on filtered sock.move.line."""
1         0         0.01              self.ensure_one()
1         0         0.09              return self.move_line_ids or self.move_line_nosuggest_ids

Total:
1         0         0.1       


2019-07-04 17:26:13,543 19288 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.02          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.03              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
1         0         0.02              move_lines = self.env['stock.move.line']
1         0         0.09              for move in self.with_context(prefetch_fields=False):
                                          move_lines |= move._get_move_lines()
                              
                                      data = MoveLine.read_group(
                                          [('id', 'in', move_lines.ids)],
                                          ['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'],
                                          lazy=False
                                      )
                              
                                      rec = defaultdict(list)
                              
                                      for d in data:
                                          rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                              
                                      for move in self:
                                          uom = move.product_uom
                                      move.quantity_done = sum(
                                          ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False)
                                          for line_uom_id, qty in rec.get(move.id, [])
                                      )

Total:
1         0         0.17      


2019-07-04 17:26:16,376 19288 INFO inertiallabs_3 odoo.tools.profiler: 
calls     queries   ms
stock.move -------------------------- /home/andrew/odoo/odoo/odoo11/addons/stock/models/stock_move.py, 250

1         0         0.01          @profile
                                  @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id', 'move_line_nosuggest_ids.qty_done')
                                  def _quantity_done_compute(self):
                                      """ This field represents the sum of the move lines `qty_done`. It allows the user to know
                                      if there is still work to do.
                              
                                      We take care of rounding this value at the general decimal precision and not the rounding
                                      of the move's UOM to make sure this value is really close to the real sum, because this
                                      field will be used in `_action_done` in order to know if the move will need a backorder or
                                      an extra move.
                                      """
1         0         0.02              MoveLine = self.env['stock.move.line']
1         0         0.02              ProductUom = self.env['product.uom']
1         0         0.01              move_lines = self.env['stock.move.line']
42        0         0.67              for move in self.with_context(prefetch_fields=False):
40        24        2489.19               move_lines |= move._get_move_lines()
                              
1         0         0.01              data = MoveLine.read_group(
1         0         5.52                  [('id', 'in', move_lines.ids)],
1         0         0.01                  ['move_id', 'product_uom_id', 'qty_done'], ['move_id', 'product_uom_id'],
1         1         77.8                  lazy=False
                                      )
                              
1         0         0.01              rec = defaultdict(list)
                              
40        0         0.16              for d in data:
39        0         0.18                  rec[d['move_id'][0]] += [(d['product_uom_id'][0], d['qty_done'])]
                              
42        0         0.38              for move in self:
40        0         0.89                  uom = move.product_uom
1         0         0.0               move.quantity_done = sum(
1         0         0.0                   ProductUom.browse(line_uom_id)._compute_quantity(qty, uom, round=False)
1         0         0.69                  for line_uom_id, qty in rec.get(move.id, [])
                                      )

Total:
1         25        2575.56   


```

Reply:
to make prefecth remove `with_context(prefect_field=False)`

Reply:
I did it.